hs_core/models.py

```diff
@@ -1750,70 +1766,494 @@ class Meta:
+            plen = len(fedpath + '/')
+            relpath = relpath[plen:]  # omit fed path
+
+            # strip resource id from path
+            if relpath.startswith(locpath):
+                plen = len(locpath)
+                relpath = relpath[plen:]  # omit local path
+            else:
+                raise ValidationError("Malformed federated resource path")
+        elif path.startswith(locpath):
+            # strip optional local path prefix
+            if test_exists and not storage.exists(path):
+                raise ValidationError("Local path does not exist in irods")
+            plen = len(locpath)
+            relpath = relpath[plen:]  # strip local prefix, omit /
+
+        # now we have folder/file. We could have gotten this from the input, or
+        # from stripping qualification folders. Note that this can contain
+        # misnamed header content misinterpreted as a folder unless one tests
+        # for existence
+        if '/' in relpath:
+            folder, base = os.path.split(relpath)
+            abspath = get_resource_path(resource, base, folder=folder)
+            if test_exists and not storage.exists(abspath):
+                raise ValidationError("Local path does not exist in irods")
+        else:
+            folder = None
+            base = relpath
+            abspath = get_resource_path(resource, base, folder=folder)
+            if test_exists and not storage.exists(abspath):
+                raise ValidationError("Local path does not exist in irods")
+
+        return folder, base
+
+    # def rename(self, new_name):
+    #     """ rename a file, setting all path variables appropriately """
+    #     pass
+
+    # def copy_irods(self, source_path, dest_path=None):
+    #     """ copy an irods file into this FileField, setting all paths appropriately """
+    #     pass
+
+    # def move_irods(self, source_path, dest_path=None):
+    #     """ move an irods file into this object, setting all paths appropriately """
+
+    # classmethods do things that query or affect all files.
+
+    @classmethod
+    def get(cls, resource, file, folder=None):
+        """
+        Get a ResourceFile record via its short path.
+        """
+        if resource.resource_federation_path:
+            return ResourceFile.objects.get(object_id=resource.id,
+                                            fed_resource_file=get_resource_path(resource,
+                                                                                file,
+                                                                                folder))
+        else:
+            return ResourceFile.objects.get(object_id=resource.id,
+                                            resource_file=get_resource_path(resource,
+                                                                            file,
+                                                                            folder))
+
+    @classmethod
+    def list(cls, resource, folder):
```

Comment:
@alvacouch This sounds to me like an instance method of the BaseResource as well.